src/Microsoft.ML.Data/Transforms/SelectColumnsTransform.cs

```diff
@@ -0,0 +1,609 @@
+        {
+            ctx.CheckAtModel(GetVersionInfo());
+            // *** Binary format ***
+            // bool: keep hidden flag
+            // int: number of added columns
+            // for each added column
+            //   string: selected column name
+            var keepHidden = ctx.Reader.ReadBoolByte();
+            var length = ctx.Reader.ReadInt32();
+            var columns = new string[length];
+            for (int i = 0; i < length; i++)
+            {
+                columns[i] = ctx.LoadNonEmptyString();
+            }
+
+            return new SelectColumnsTransform(env, keepHidden, columns);
+        }
+
+        // Factory method for SignatureLoadDataTransform.
+        public static IDataView Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(ctx, nameof(ctx));
+            SelectColumnsTransform transform;
+
+            // Determine which version of the transform is being loaded.
+            if (CheckModelVersion(ctx, GetDropVersionInfo()))
+            {
+                transform = LoadDropColumnsTransform(env, ctx, input);
+            }
+            else if (CheckModelVersion(ctx, GetChooseVersionInfo()))
+            {
+                transform = LoadChooseColumnsTransform(env, ctx, input);
+            }
+            else
+            {
+                transform = Create(env, ctx);
+            }
+
+            return transform.Transform(input);
+        }
+
+        // Factory method for SignatureDataTransform.
+        private static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(args, nameof(args));
+            var transform = new SelectColumnsTransform(env, args.KeepHidden, args.Columns);
+            return new SelectColumnsDataTransform(env, transform, new Mapper(transform, input.Schema), input);
+        }
+
+        public void Save(ModelSaveContext ctx)
+        {
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            ctx.Writer.WriteBoolByte(KeepHidden);
+            var length = _selectedColumns.Length;
+            ctx.Writer.Write(length);
+            for (int i = 0; i < length; i++)
+                ctx.SaveNonEmptyString(_selectedColumns[i]);
+        }
+
+        public Schema GetOutputSchema(Schema inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            return new Mapper(this, inputSchema).Schema;
+        }
+
+        public IRowToRowMapper GetRowToRowMapper(Schema inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            return new SelectColumnsDataTransform(_host, this,
+                                                  new Mapper(this, inputSchema),
+                                                  new EmptyDataView(_host, Schema.Create(inputSchema)));
```

Comment:
actually if you look at `Schema.Create`, you'll see that it will just return `inputSchema`, so let's not call the method at all

---
In reply to: [226045960](https://github.com/dotnet/machinelearning/pull/1269#discussion_r226045960) [](ancestors = 226045960)